packages/arb-rpc-node/batcher/sequencerBatcher.go

```diff
@@ -202,6 +256,106 @@ func (b *SequencerBatcher) SendTransaction(_ context.Context, startTx *types.Tra
 	l2Message := message.NewSafeL2Message(batch)
 	seqMsg := message.NewInboxMessage(l2Message, b.sequencer, new(big.Int).Set(msgCount), big.NewInt(0), b.latestChainTime.Clone())
 
+	logCount, err := b.db.GetLogCount()
+	if err != nil {
+		return err
+	}
+
+	txBatchItem := inbox.NewSequencerItem(totalDelayedCount, seqMsg, prevAcc)
+	err = core.DeliverMessagesAndWait(b.db, prevAcc, []inbox.SequencerBatchItem{txBatchItem}, []inbox.DelayedMessage{}, nil)
+	if err != nil {
+		return err
+	}
+	core.WaitForMachineIdle(b.db)
+
+	var sequencedTxs []*types.Transaction
+	var sequencedBatchItems []inbox.SequencerBatchItem
+
+	newLogCount, err := b.db.GetLogCount()
+	if err != nil {
+		return err
+	}
+	txLogs, err := b.db.GetLogs(logCount, new(big.Int).Sub(newLogCount, logCount))
+	if err != nil {
+		return err
+	}
+	txResults, err := txLogsToResults(txLogs)
+	if err != nil {
+		return err
+	}
+
+	successCount := 0
+	for _, tx := range batchTxs {
+		if txResults[common.NewHashFromEth(tx.Hash())] {
+			successCount++
+		}
+	}
+	if successCount == len(batchTxs) {
+		sequencedTxs = batchTxs
+		prevAcc = txBatchItem.Accumulator
+		for _, c := range resultChans {
+			c <- nil
+		}
+	} else {
+		// Reorg to before we processed the batch and re-process the messages individually
+		err = core.DeliverMessagesAndWait(b.db, prevAcc, nil, nil, msgCount)
+		if err != nil {
+			return err
+		}
+		core.WaitForMachineIdle(b.db)
+		if successCount == 0 {
+			// All of the transactions failed
+			for _, c := range resultChans {
+				c <- errors.New("transaction errored")
+			}
+			return <-startResultChan
+		}
+		// At least one of the transactions failed and one of the transactions succeeded
+		for i, tx := range batchTxs {
+			txHash := common.NewHashFromEth(tx.Hash())
+			if !txResults[txHash] {
+				continue
+			}
+			l2Msg := message.NewCompressedECDSAFromEth(tx)
+			batch, err = message.NewTransactionBatchFromMessages([]message.AbstractL2Message{l2Msg})
+			if err != nil {
+				return err
+			}
+			l2Message := message.NewSafeL2Message(batch)
+			seqMsg := message.NewInboxMessage(l2Message, b.sequencer, new(big.Int).Set(msgCount), big.NewInt(0), b.latestChainTime.Clone())
+			txBatchItem := inbox.NewSequencerItem(totalDelayedCount, seqMsg, prevAcc)
+			err = core.DeliverMessagesAndWait(b.db, prevAcc, []inbox.SequencerBatchItem{txBatchItem}, []inbox.DelayedMessage{}, nil)
+			if err != nil {
+				return err
+			}
+			newLogCount, err = b.db.GetLogCount()
+			if err != nil {
+				return err
+			}
+			txLogs, err = b.db.GetLogs(logCount, new(big.Int).Sub(newLogCount, logCount))
+			if err != nil {
+				return err
+			}
+			newTxResults, err := txLogsToResults(txLogs)
+			if err != nil {
+				return err
+			}
+			if !newTxResults[txHash] {
+				err = core.DeliverMessagesAndWait(b.db, prevAcc, nil, nil, msgCount)
+				if err != nil {
+					return err
+				}
+				resultChans[i] <- errors.New("transaction errored")
```

Comment:
We need to clean up the errors a bit. I think we can use https://github.com/OffchainLabs/arbitrum/blob/cc8d690934ba3bb7507bc72c34c35d363fc23b70/packages/arb-rpc-node/web3/eth.go#L208 with `ganacheMode` false, but it might need some refactoring since as-is there'd be a dependency loop. Could probably move the function to `evm`.

I think it'd also make sense